src/cinder/gl/GlslProg.cpp

```diff
@@ -697,10 +718,20 @@ void GlslProg::loadShader( const string &shaderSource, const fs::path &shaderPat
 			}
 			log += "\n" + ss.str();
 		}
-#endif		
+#endif
+		// Since the GlslProg destructor will not be called after we throw, we must delete all
+		// owned GL objects here to avoid leaking. Any other attached shaders will be cleaned up
+		// when the program is deleted
+		glDeleteShader( handle );
+		glDeleteProgram( mHandle );
```

Comment:
That thought had actually crossed my mind after writing the previous comment. I think it would be fine to pass in an empty string to loadShader(), and if so, it just returns 0. That method is internal, so sometimes for things like this I just write a little comment above it in the cpp for the next fellow who comes along.